romeo/src/state.rs

```diff
@@ -469,13 +473,44 @@ impl State {
 			State::Initialized {
 				deposits,
 				withdrawals,
+				stacks_block_height,
 				..
 			} => {
 				let deposit_tasks = deposits.iter_mut().filter_map(|deposit| {
 					match deposit.mint.as_mut() {
 						None => {
-							deposit.mint = Some(TransactionRequest::Created);
-							Some(Task::CreateMint(deposit.info.clone()))
+							// We often receive the deposit before the
+							// transaction is actually mined. By scheduling the
+							// transaction for a block later than the current
+							// one we make ourselves resilient to mining delays
+							// without complex logic.
+							let scheduled_block_height = *stacks_block_height
+								+ MINT_TRANSACTION_DELAY_BLOCKS;
+							deposit.mint =
+								Some(TransactionRequest::Scheduled {
+									block_height: scheduled_block_height,
+								});
+							info!("Scheduled deposit {} for minting on stacks block height {}.",
+								deposit.info.txid, scheduled_block_height);
+							None
+						}
+						Some(TransactionRequest::Scheduled {
+							block_height,
+						}) => {
+							if stacks_block_height.ge(&block_height) {
```

Comment:
What happens if the stack blocks are full and the mint is not included at the expected height?

Reply:
Then we're SOL, this is not an ideal solution but it's one that'll fix the issue 99/100 times.

Reply:
There's a world in which we hit the transaction with the clarity read-only function `verify-txid-exists-on-burn-chain` and wait for each block until that verification passes - that's the *right* way to do this.

We said we'd have on testnet a while ago so it's better to make a fast decent fix that will likely solve the problem than introduce complexity. Especially since this code isn't going to be built upon for the actual sBTC release.

If we have this problem again we can refer to these comments.

Reply:
SOL?

Fine with me. I wanted an answer for the record in case I missed something.

Reply:
We only want to prevent broadcasting too early. We have an extra event for inclusion once the tx was confirmed.

I don't see a problem here.